app/views/groups/_group_card.html.erb

```diff
@@ -1,38 +1,46 @@
-<div class="card">
+<div class="card" id="<%= dom_id(group) %>">
 
   <div class="card-header">
     <h6 class="card-title">
-      <%= link_to group.group_name, group %>    
+      <%= link_to group.group_name, group, id: dom_id(group)+"_group_name" %>    
         <span>
-          <%= link_to new_membership_path, class: "btn btn-link btn-sm text-muted" do %>
+
+          <%= link_to new_membership_path(:group_id => group.id), class: "btn btn-link btn-sm text-muted", remote: true do %>
           <i class="bi bi-person-plus-fill"></i>
           <% end %>
-
-          <%= link_to edit_group_path(group), class: "btn btn-link btn-sm text-muted" do %>
+          
+          <%= link_to edit_group_path(group), class: "btn btn-link btn-sm text-muted", remote: true do %>
             <i class="bi bi-pencil-fill"></i>
           <% end %>
 
-          <%= link_to group, method: :delete, class: "btn btn-link btn-sm text-muted" do %>
+          <%= link_to group, method: :delete, class: "btn btn-link btn-sm text-muted", remote: true do %>
             <i class="bi bi-trash-fill"></i>
           <% end %>
         </span>
     </h6>
   </div>
 
   <div>
-    <details > <!--TODO: I need to change this for a modal or dropdown-->
-      <summary> Show contacts </summary>
-      <ul class="list-group list-group-flush">
-        <% group.contacts.order(:last_name).each do |contact|%>
-        <li class="list-group-item">
-          <%= link_to "#{contact.name} #{contact.last_name}", contact%> uses <%= contact.communication_channel%> 
-            <span class="badge float-right"> 
-                <%= link_to membership_path(Membership.find_by(group_id: group.id, contact_id: contact.id).id), method: :delete, class: "btn btn-link btn-sm text-muted" do %>              <i class="bi bi-trash-fill"></i>
-              <% end %>
-            </span>
-        </li>
-        <%end%>
-      </ul>
-    </details>
+
+      <button class="btn btn-primary btn-sm" type="button" data-toggle="collapse" data-target="#<%= dom_id(group) %>_contacts" aria-expanded="false" aria-controls="<%= dom_id(group) %>_contacts">
+        Show contacts
+      </button>
+
+    <div class="collapse" id="<%= dom_id(group) %>_contacts">
+      <div class="card card-body">
+        <ul class="list-group list-group-flush">
+          <% group.contacts.order(:last_name).each do |contact|%>
+          <li class="list-group-item" id="membership_<%= Membership.where(group_id: group.id, contact_id: contact.id).ids.first %>">
+            <%= link_to "#{contact.name} #{contact.last_name}", contact%> uses <%= contact.communication_channel%> 
+              <span class="badge float-right"> 
+                <%= link_to membership_path(Membership.where(group_id: group.id, contact_id: contact.id).ids), method: :delete, remote: true, class: "btn btn-link btn-sm text-muted" do %>
```

Comment:
Does this delete link work? 

My concern is that `Membership.where(group_id: group.id, contact_id: contact.id).ids` evaluates to some Array object, which means the link_to looks like

```erb
<%= link_to membership_path([...]) %>
```

which I've never done before.

---

Separately, you already do the work of finding the membership record above. You could save that in a variable (like `membership`), and then you [could re-use it in this link_to](https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to)

```erb
                <%= link_to membership, method: :delete, remote: true, class: "btn btn-link btn-sm text-muted" do %>
```

What do you think?

Reply:
It works! But I guess it is because rails is great. I am going to change it using `find_by` as I would like to separate how it reads for the id and for the path